sys/kern/thread.c

```diff
@@ -131,6 +131,11 @@ void thread_delete(thread_t *td) {
   pool_free(P_THREAD, td);
 }
 
+/*
+ * Instrumentation in this function makes KCSAN crash. I don't know the cause,
+ * but the function is so simple that there is no need to instrumentate it
+ * anyway.
+ */
```

Comment:
Maybe it crashes for the same reason as `preemt_disabled()` - infinite recursion. Notice that `preempt_disabled()` calls `thread_self()`.

Reply:
Oh right, I missed that.